src/components/Contacts/ContactDetails/ContactDetailsTab/PartnerAccounts/ContactDetailsPartnerAccounts.test.tsx

```diff
@@ -38,72 +43,83 @@ jest.mock('notistack', () => ({
-          displayName: 'donor-1',
-          accountNumber: 'donor-1',
-          organization: {
-            id: 'org1',
-            name: 'org1',
+const ContactDonorAccountsMock: ContactDonorAccountsQuery = {
+  contact: {
+    id: contactId,
+    contactDonorAccounts: {
+      nodes: [
+        {
+          id: 'donor1',
+          donorAccount: {
+            id: 'account1',
+            displayName: 'donor-1',
+            accountNumber: 'accountNumber-1',
+            organization: {
+              id: 'org1',
+              name: 'org1',
+            },
           },
         },
-      },
-      {
-        id: 'donor2',
-        donorAccount: {
-          id: 'account2',
-          displayName: 'donor-2',
-          accountNumber: 'donor-2',
-          organization: {
-            id: 'org2',
-            name: 'org2',
+        {
+          id: 'donor2',
+          donorAccount: {
+            id: 'account2',
+            displayName: 'donor-2',
+            accountNumber: 'accountNumber-2',
+            organization: {
+              id: 'org2',
+              name: 'org2',
+            },
           },
         },
-      },
-    ],
+      ],
+    },
   },
 };
 
+const Components = () => (
+  <SnackbarProvider>
+    <TestRouter router={router}>
+      <GqlMockedProvider<{
+        ContactDonorAccounts: ContactDonorAccountsQuery;
+        GetAccountListSalaryOrganization: GetAccountListSalaryOrganizationQuery;
+      }>
+        mocks={{
+          ContactDonorAccounts: ContactDonorAccountsMock,
+          GetAccountListSalaryOrganization: {
+            accountList: { salaryOrganizationId: 'organizationId' },
+          },
+        }}
+        onCall={mutationSpy}
+      >
+        <ThemeProvider theme={theme}>
+          <ContactDetailProvider>
+            <ContactDetailsPartnerAccounts
+              contactId={contactId}
+              accountListId={accountListId}
+            />
+          </ContactDetailProvider>
+        </ThemeProvider>
+      </GqlMockedProvider>
+    </TestRouter>
+  </SnackbarProvider>
+);
+
 describe('ContactDetailsPartnerAccounts', () => {
   it('should render donor account', async () => {
-    const { getByText } = render(
-      <SnackbarProvider>
-        <TestRouter router={router}>
-          <GqlMockedProvider>
-            <ThemeProvider theme={theme}>
-              <ContactDetailProvider>
-                <ContactDetailsPartnerAccounts contact={contact} />
-              </ContactDetailProvider>
-            </ThemeProvider>
-          </GqlMockedProvider>
-        </TestRouter>
-      </SnackbarProvider>,
-    );
+    const { findByText, getByText, getAllByText } = render(<Components />);
 
-    expect(getByText('donor-1')).toBeInTheDocument();
-    expect(getByText('donor-2')).toBeInTheDocument();
+    expect(await findByText('accountNumber-1')).toBeInTheDocument();
+    expect(getByText('accountNumber-2')).toBeInTheDocument();
+
+    expect(getAllByText('Account No:').length).toEqual(2);
```

Comment:
Optional, but I like using `toHaveLength`.

```suggestion
    expect(getAllByText('Account No:')).toHaveLength(2);
```